File: app/demo/src/main/java/com/example/demo/web/dto/DatabaseEnum.java
package com.example.demo.web.dto;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum DatabaseEnum {
    CLICKHOUSE("CLICKHOUSE"), MYSQL("MYSQL"), BOTH("BOTH");

    private final String text;

    DatabaseEnum(final String text) {
        this.text = text;
    }

    @Override
    public String toString() {
        return text;
    }

    public String getText(){return this.text;}

    @JsonCreator
    public static DatabaseEnum fromText(String text){
        for(DatabaseEnum d : DatabaseEnum.values()){
            if(d.getText().equals(text)){
                return d;
            }
        }
        throw new IllegalArgumentException();
    }
}

File: app/webapp/src/app/enum/database-enum.ts
export enum DatabaseEnum {
  MYSQL = "MYSQL",
  CLICKHOUSE = "CLICKHOUSE",
  BOTH = "BOTH"
}


export const getDabaseEnum = (value: string): DatabaseEnum => {
  switch (value) {
    case DatabaseEnum.MYSQL:
      return DatabaseEnum.MYSQL;
    case DatabaseEnum.CLICKHOUSE:
      return DatabaseEnum.CLICKHOUSE;
    default:
      return DatabaseEnum.BOTH;
  }
};
